crates/cheatnet/src/state.rs

```diff
@@ -158,6 +159,24 @@ pub struct CallTrace {
     pub nested_calls: Vec<Rc<RefCell<CallTrace>>>,
     pub result: CallResult,
     pub vm_trace: Option<Vec<TraceEntry>>,
+    pub node_type: NodeType,
+}
+
+#[derive(Default, Copy, Clone)]
+pub enum NodeType {
+    #[default]
+    Regular,
+    Phantom,
+}
+
+impl NodeType {
+    #[must_use]
+    pub fn is_visible_to_user(&self) -> bool {
```

Comment:
Not sure what this function does, but it seems like mixing presentation layer with data definition layer.

Reply:
Definetely shouldn't be in cheatnet

Reply:
Will be gone with enum introduction